### ISSUE TITLE
Error after PG and Bao setup

### ISSUE THREAD

**Issue by LumingSun** [learnedsystems/BaoForPostgreSQL#5]
Hi Ryan, great work!
I encountered some problem when setting the Bao extension.
PostgreSQL version: 12.2
The database has been created and the extension is installed without problems. The bao server has been started and ```SET enable_bao TO on;``` is OK. But when I run ```EXPLAIN SELECT count(*) FROM title;```, I got 
```
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
```
And I tested that if I don't add ```shared_preload_libraries = 'pg_bao'``` to the pg config, everything is OK. (But of course I can not use Bao extension). And with the above config, the problem exists even ```enable_bao``` is off.
Is there anything I missed?

**Comment by LumingSun** [comment]
And here is the log
```
 2021-11-18 15:37:36.223 CST [48887] LOG:  server process (PID 49950) was terminated by signal 6: 已放弃
 2021-11-18 15:37:36.223 CST [48887] DETAIL:  Failed process was running: EXPLAIN SELECT count(*) FROM title;
 2021-11-18 15:37:36.223 CST [48887] LOG:  terminating any other active server processes
 2021-11-18 15:37:36.224 CST [48930] WARNING:  terminating connection because of crash of another server process
 2021-11-18 15:37:36.224 CST [48930] DETAIL:  The postmaster has commanded this server process to roll back the current transaction and exit, because another server process exited abnormally and possibly corrupted shared memory.
 2021-11-18 15:37:36.224 CST [48930] HINT:  In a moment you should be able to reconnect to the database and repeat your command.
 2021-11-18 15:37:36.228 CST [49965] FATAL:  the database system is in recovery mode
 2021-11-18 15:37:36.297 CST [48887] LOG:  all server processes terminated; reinitializing
 2021-11-18 15:37:39.137 CST [50075] LOG:  database system was interrupted; last known up at 2021-11-18 15:33:25 CST
 2021-11-18 15:37:39.171 CST [50075] LOG:  database system was not properly shut down; automatic recovery in progress
 2021-11-18 15:37:39.173 CST [50075] LOG:  redo starts at 2/19A4A6A8
 2021-11-18 15:37:39.173 CST [50075] LOG:  invalid record length at 2/19A4A6E0: wanted 24, got 0
 2021-11-18 15:37:39.173 CST [50075] LOG:  redo done at 2/19A4A6A8
 2021-11-18 15:37:39.791 CST [48887] LOG:  database system is ready to accept connections```

**Comment by LumingSun** [comment]
Solved according to #3
